From 5849c5808a721893da4decb87aa9a8b751a06edd Mon Sep 17 00:00:00 2001
Message-Id: <5849c5808a721893da4decb87aa9a8b751a06edd.1421709247.git.chang-joon.lee@intel.com>
In-Reply-To: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
References: <3650eb44734ad4fbe098313349da8185595a20d8.1421709247.git.chang-joon.lee@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Sun, 19 Oct 2014 14:58:54 +0530
Subject: [PATCH 57/75] MUST_REBASE [VPG]: adf/intel/vlv: Send Fake signal to
 user when we disable/enable dpst via debugfs

Send a fake signal to user, so that the user can be notified
to reset the dpst context.

MUST_REBASE: Since Google ADF framework is not in upstream and also
we cannot have two display drivers, other one being i915, we need to
work on i915 and ADF convergence path before it can be upstreamed

For: GMINL-3547
Change-Id: Ic7da79146112a467d23628aafa82efb85829f898
Signed-off-by: Deepak S <deepak.s@intel.com>
---
 drivers/video/adf/intel/core/vlv/vlv_dpst.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/video/adf/intel/core/vlv/vlv_dpst.c b/drivers/video/adf/intel/core/vlv/vlv_dpst.c
index 347cef4..73baa81 100644
--- a/drivers/video/adf/intel/core/vlv/vlv_dpst.c
+++ b/drivers/video/adf/intel/core/vlv/vlv_dpst.c
@@ -625,6 +625,7 @@ int
 vlv_dpst_set_kernel_disable(struct vlv_dc_config *vlv_config, bool disable)
 {
 	struct intel_dc_config *config = g_config;
+	struct intel_pipe *pipe = g_config->pipes[vlv_config->dpst.pipe];
 	int ret = 0;
 
 	mutex_lock(&vlv_config->dpst.ioctl_lock);
@@ -642,6 +643,11 @@ vlv_dpst_set_kernel_disable(struct vlv_dc_config *vlv_config, bool disable)
 
 	mutex_unlock(&vlv_config->dpst.ioctl_lock);
 
+	/* Send a fake signal to user, so that the user can be notified
+	 * to reset the dpst context, to avoid any mismatch of blc_adjusment
+	 * between user and kernel on resume. */
+	vlv_dpst_irq_handler(pipe);
+
 	return ret;
 }
 
-- 
1.7.9.5

